diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59121ec..f17b85c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,7 +176,7 @@ add_subdirectory(plugins/Lomiri/Thumbnailer.0.1)
 if (${ENABLE_UBUNTU_COMPAT})
     add_subdirectory(plugins/Ubuntu/Thumbnailer.0.1)
 endif()
-add_subdirectory(tests)
+#add_subdirectory(tests)
 add_subdirectory(include)
 add_subdirectory(man)
 #add_subdirectory(doc)
