From: Bardia Moshiri <fakeshell@bardia.tech>
diff -Npur a/data/lomiri-indicator-network.service.in b/data/lomiri-indicator-network.service.in
--- a/data/lomiri-indicator-network.service.in	2023-05-24 18:27:58.002376294 +0000
+++ b/data/lomiri-indicator-network.service.in	2023-05-24 18:28:30.675172314 +0000
@@ -7,7 +7,7 @@ Wants=lomiri-indicator-network-secret-ag
 
 [Service]
 BusName=com.lomiri.connectivity1
-ExecStart=@NETWORK_SERVICE@
+ExecStart=/usr/lib/indicator-network/indicator-network-service
 Restart=on-failure
 
 [Install]
