diff --git a/data/lomiri-indicator-network.service.in b/data/lomiri-indicator-network.service.in
index 9d547c8..02928e1 100644
--- a/data/lomiri-indicator-network.service.in
+++ b/data/lomiri-indicator-network.service.in
@@ -1,13 +1,11 @@
 [Unit]
 Description=Indicator Network Backend
-PartOf=graphical-session.target
-After=indicators-pre.target
+PartOf=ayatana-indicators.target
 Wants=lomiri-indicator-network-secret-agent.service
 
 [Service]
-ExecStart=@NETWORK_SERVICE@
+ExecStart=/usr/lib/indicator-network/indicator-network-service
 Restart=on-failure
 
 [Install]
-# FIXME: use the Ayatana's target once we're fully an Ayatana indicator
-WantedBy=indicators-pre.target
+WantedBy=ayatana-indicators.target
