From 4063219923d8019896b1b423301866bd056a1676 Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Tue, 8 Sep 2020 21:29:00 +0200
Subject: [PATCH] Update for libebook >= 3.33.2

---
 lib/qindividual.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/qindividual.cpp b/lib/qindividual.cpp
index 4bcde46..be984e4 100644
--- a/lib/qindividual.cpp
+++ b/lib/qindividual.cpp
@@ -1179,7 +1179,7 @@ bool QIndividual::markAsDeleted()
                 e_vcard_attribute_add_value(attr, currentDate.toUtf8().constData());
             }
 
-            e_book_client_modify_contact_sync(E_BOOK_CLIENT(client), c, NULL, &error);
+            e_book_client_modify_contact_sync(E_BOOK_CLIENT(client), c, E_BOOK_OPERATION_FLAG_NONE, NULL, &error);
             if (error) {
                 qWarning() << "Fail to update EDS contact:" << error->message;
                 g_error_free(error);
@@ -1798,7 +1798,7 @@ void QIndividual::setExtendedDetails(FolksPersona *persona,
                 }
             }
 
-            e_book_client_modify_contact_sync(E_BOOK_CLIENT(client), c, NULL, &error);
+            e_book_client_modify_contact_sync(E_BOOK_CLIENT(client), c, E_BOOK_OPERATION_FLAG_NONE, NULL, &error);
             if (error) {
                 qWarning() << "Fail to update EDS contact:" << error->message;
                 g_error_free(error);
-- 
2.28.0

